### PR TITLE
Fix(GH) : account token component not finding correct account

### PR DESCRIPTION
**Pull request by connorivy**
## Description & motivation

Account token component was broken by this PR https://github.com/specklesystems/speckle-sharp/pull/3141 which changed the behavior of accounts

This PR applies the same fix that was used for the other components that depended on the account component.

## Checklist:

- [x] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [x] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [x] My commits are related to the pull request and do not amend unrelated code or documentation.
- [x] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

File: ConnectorGrasshopper/ConnectorGrasshopperShared/Accounts/Accounts.GetAccountToken.cs [hunk 1]
```diff
@@ -46,12 +47,7 @@ public Accounts_GetAccountToken()
 
   protected override void RegisterInputParams(GH_InputParamManager pManager)
   {
-    pManager.AddTextParameter(
-      "Account",
-      "A",
-      "Account to get the auth token from. Expects the `userId`",
-      GH_ParamAccess.item
-    );
+    pManager.AddParameter(new SpeckleAccountParam());
```
Comment (clairekuang) [inline ConnectorGrasshopper/ConnectorGrasshopperShared/Accounts/Accounts.GetAccountToken.cs:50]: This changes the input parameter type, which would be a breaking change for users with GH scripts with this component that are passing in a text parameter. Are we okay with this @AlanRynne ?

Comment (clairekuang) [reply ConnectorGrasshopper/ConnectorGrasshopperShared/Accounts/Accounts.GetAccountToken.cs:50]: Nevermind, looks like the input node was an `Accounts` node which previously passed the user id as the output now passes the `Account` object. Looks good to go ✌️

**Review by clairekuang — APPROVED** [review]

**Review by AlanRynne — CHANGES_REQUESTED** [review]
These changes do the trick except for updating both name and description of the account input. Built with this branch, i tested an opening a file with an existing "Get Token From Account" component

Screenshot 2024-02-14 at 09 58 11

While the a newly placed component will look like this:

Screenshot 2024-02-14 at 09 58 15

I'm not sure this is a big enough issue to request changing this.
The solution would involve:
- Leaving the original component intact
- Creating a brand new component with these changes
- Create a class that implements IGH_UpgradeObject to define what is involved in "upgrading" an existing component to the new one.

This would also have the side-effect of forcing the users to upgrade their existing nodes in each file (which is Grasshopper recommended approach) as this will only affect descriptions and names of pre-existing components, but the components **will work as expected**

**Review by AlanRynne — APPROVED** [review]
Approving after discussion during standup.

Opened ticket to polish off the UI quirks in my comment https://spockle.atlassian.net/browse/CNX-8923

**Pull request merged**